apps/backend/src/experiments/core/repositories/experiment.repository.ts

```diff
@@ -51,35 +52,87 @@ export class ExperimentRepository {
     return tryCatch(() => {
       // Start with a base query builder
       const query = this.database.select(experimentFields).from(experiments);
-
-      // Apply filters based on the filter type
-      switch (filter) {
-        case "my":
-          return query.where(eq(experiments.createdBy, userId));
-
-        case "member":
-          return query
-            .innerJoin(
-              experimentMembers,
-              eq(experiments.id, experimentMembers.experimentId),
-            )
-            .where(eq(experimentMembers.userId, userId));
-
-        case "related":
-          return query
-            .leftJoin(
-              experimentMembers,
-              eq(experiments.id, experimentMembers.experimentId),
-            )
-            .where(
-              or(
+      
+      // Build different queries based on combined filter and status
+      if (filter && status) {
```

Comment:
You should be able to clean this up by just doing the status at the end

```
    if (status) {
        query = query.where(eq(experiments.status, status));
      }
```
or at least I am 95% sure needs to be tried out